src/ast-to-react.js

```diff
@@ -2,257 +2,250 @@
 
 const React = require('react')
 const ReactIs = require('react-is')
+const convert = require('unist-util-is/convert')
+const svg = require('property-information/svg')
+const find = require('property-information/find')
+const hastToReact = require('property-information/hast-to-react.json')
+const spaces = require('space-separated-tokens')
+const commas = require('comma-separated-tokens')
+const style = require('style-to-object')
+
+exports.hastToReact = toReact
+exports.hastChildrenToReact = childrenToReact
+
+const element = convert('element')
+const text = convert('text')
+
+const own = {}.hasOwnProperty
+
+function childrenToReact(context, node) {
+  const children = []
+  let childIndex = -1
+  let child
+
+  while (++childIndex < node.children.length) {
+    child = node.children[childIndex]
+
+    if (element(child)) {
+      children.push(toReact(context, child, childIndex, node))
+    } else if (text(child)) {
+      children.push(child.value)
+    } else if (child.type === 'raw' && !context.options.skipHtml) {
+      // Default behavior is to show (encoded) HTML.
+      children.push(child.value)
+    }
+  }
```

Comment:
There are some possible holes (comments, raw w/o `skipHtml`). I like the explicitness of loops such as this

Reply:
Fair enough,
It could work with a `filter` and a `map`.
I tend to avoid `push` inside loop statements.
Even as far as having a eslint rule for it :sweat_smile: 
```json
{
    "rules": {
        "no-restricted-syntax": [
            "ForStatement CallExpression MemberExpression[property.name=push]",
            "ForInStatement CallExpression MemberExpression[property.name=push]",
            "ForOfStatement CallExpression MemberExpression[property.name=push]",
            "WhileStatement CallExpression MemberExpression[property.name=push]"
        ]
    }
}
```
But keeping the looping construct could be fine in this context.